packages/opentelemetry/src/spanExporter.ts

```diff
@@ -159,13 +171,13 @@ function createTransactionForOtelSpan(span: ReadableSpan): Transaction {
     status: mapStatus(span),
     startTimestamp: convertOtelTimeToSeconds(span.startTime),
     metadata: {
-      sampleRate: span.attributes[SEMANTIC_ATTRIBUTE_SENTRY_SAMPLE_RATE] as number | undefined,
+      ...dropUndefinedKeys({
```

Comment:
here I think it is important, because `dynamicSamplingContext` may be `undefined` in which case we don't want to set it, because later in the pipeline if you have e.g. this:

```
metadata: {
  dynamicSamplingContext: xx,
  ...event.metadata
}
```
this would overwrite `xx` even if it is `undefined`, I think?